app/src/main/java/com/mwalasiak/pyramid/LevelsActivity.kt

```diff
@@ -40,94 +48,72 @@ class LevelsActivity : AppCompatActivity() {
 
-    private fun fillSeventhLevel() {
-        fillLevels(SIXTH_LEVEL_KEY, SEVENTH_LEVEL_KEY, binding.seventhLevel.id)
+    private fun fillThirdLevel() {
+        fillLevel(SECOND_LEVEL_KEY, THIRD_LEVEL_KEY, binding.thirdLevel)
     }
 
-    private fun fillEighthLevel() {
-        fillLevels(SEVENTH_LEVEL_KEY, EIGHTH_LEVEL_KEY, binding.eighthLevel.id)
+    private fun fillSecondLevel() {
+        fillLevel(FIRST_LEVEL_KEY, SECOND_LEVEL_KEY, binding.secondLevel)
     }
 
-    private fun fillNinthLevel() {
-        fillLevels(EIGHTH_LEVEL_KEY, NINTH_LEVEL_KEY, binding.ninethLevel.id)
-    }
+    private fun fillFirstLevel() {
+        if (verifySharedPref(TMP_LIST) && binding.enterNumber.text.isNotEmpty()) {
+            val text = getList(TMP_LIST)
+            setList(text, FIRST_LEVEL_KEY)
+            binding.firstLevel.text = getList(FIRST_LEVEL_KEY)?.replace(",", " ")
+            binding.enterNumber.clear()
+            binding.addLevels.isEnabled = false
+        } else
+            binding.firstLevel.text = EMPTY_LIST_TEXT
 
-    private fun fillTenthLevel() {
-        fillLevels(NINTH_LEVEL_KEY, TENTH_LEVEL_KEY, binding.tenthLevel.id)
     }
 
-    private fun fillLevels(previousLevel:String, currentLevel:String, id: Int) {
-        if (verifySharedPref(previousLevel) && getTextView(binding.enterNumber.id).text.isNotEmpty()){
+    private fun fillLevel(previousLevel:String, currentLevel:String, view: TextView) {
+        if (verifySharedPref(previousLevel) && binding.enterNumber.text.isNotEmpty()){
             val tmpList = compareLevels(getList(previousLevel), getList(TMP_LIST))
             setList(tmpList, currentLevel)
-            getTextView(id).text = getList(currentLevel)?.replace(",", " ")
+            view.text = getList(currentLevel)?.replace(",", " ")
             binding.saveNumbers.isEnabled = true
             binding.addLevels.isEnabled = false
         }else
-            getTextView(id).text = EMPTY_LIST_TEXT
+            view.text = EMPTY_LIST_TEXT
 
     }
 
-    private fun getTextView(id: Int) = binding.root.findViewById<TextView>(id)
-
-    private fun saveNumbersToTmpList() {
-        val textFromInput = binding.enterNumber.text.toString()
-        if (textFromInput.isNotEmpty()){
-            setList(textFromInput, TMP_LIST)
-            binding.saveNumbers.isEnabled = false
-            binding.addLevels.isEnabled = true
-        }else
-            setToastMessage()
-    }
-
-    private fun setList(stringWithNumbers: String?, listKey: String) {
-        val editor = setSharedPreferences().edit()
-        editor.putString(listKey, stringWithNumbers)
-        editor.apply()
-    }
-
-    private fun getList(listName: String): String? {
-        return setSharedPreferences().getString(listName, null)
-    }
-
     private fun compareLevels(
         listA: String?,
         listB: String?,
     ): String {
 
-        if (listA!!.isNotEmpty()){
+        if (listA!!.isNotEmpty() && listA != EMPTY_LIST_TEXT){
```

Comment:
Please try avoiding `!!` in production code at all cost. If you are SURE that this list, will not be empty, then don't allow to pass empty list to the function :) `!!` should only be available in the test code. In here, we should do a workaround for each case (for example not doing something, or fallbacking to empty lists etc.)

Reply:
But in this case list a is a string and it might be empty, listA is a wrong name which need to be changed